packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -225,6 +254,49 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
       pointerSignalResolver.resolve(event);
     }
   }
+
+  void _scheduleFrameCallback() {
+    if (_frameCallbackScheduled) {
+      return;
+    }
+    final SchedulerBinding? instance = scheduler ??= SchedulerBinding.instance;
+    assert(instance != null);
+    if (instance != null) {
+      _frameCallbackScheduled = true;
+      instance.scheduleFrameCallback((_) {
+        _frameCallbackScheduled = false;
+        _sampleTime = instance.currentSystemFrameTimeStamp + samplingOffset;
+        if (!locked) {
+          _flushPointerEventQueue();
+        }
+      });
+    }
+  }
+
+  // Flag to track if a frame callback has been scheduled.
+  bool _frameCallbackScheduled = false;
+
+  // Current sample time for resampling.
+  Duration _sampleTime = const Duration();
+
+  // Resamplers used to filter incoming touch events when resampling
+  // is enabled.
+  final Map<int, PointerEventResampler> _resamplers =
+      <int, PointerEventResampler>{};
+
+  /// Enable pointer event resampling by setting this to `true`.
+  ///
+  /// Resampling results in smoother event processing on devices
+  /// without high frequency sensors at the cost of adding some latency.
```

Comment:
It would be nice to add cases like `or when the high input frequency is not a multiple of the display frequency (e.g., 120Hz input and 90Hz display)`.

Reply:
Done